### PR TITLE
Fix fix all exception with duplicate Program declaration in top level program

### PR DESCRIPTION
**Pull request by dibarbet**
Resolves https://github.com/dotnet/vscode-csharp/issues/6611

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AddAccessibilityModifiers
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsAddAccessibilityModifiers)]
+    public class AddAccessibilityModifiersFixAllTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
```
Comment (mavasani) [inline src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs:21]: Let us avoid using the legacy test framework for these new tests. See https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Tests/AddBraces/AddBracesFixAllTests.cs for example FixAll tests based on the new Microsoft.CodeAnalysis.Testing library.

File: src/Analyzers/CSharp/CodeFixes/AddAccessibilityModifiers/CSharpAddAccessibilityModifiersCodeFixProvider.cs [hunk 2]
```diff
@@ -15,7 +16,7 @@ namespace Microsoft.CodeAnalysis.CSharp.AddAccessibilityModifiers
     internal class CSharpAddAccessibilityModifiersCodeFixProvider : AbstractAddAccessibilityModifiersCodeFixProvider
     {
         [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
```
Comment (mavasani) [inline src/Analyzers/CSharp/CodeFixes/AddAccessibilityModifiers/CSharpAddAccessibilityModifiersCodeFixProvider.cs:19]: You should be able to revert these changes once you move the tests to the new testing library.

**Comment by arkalyanms** [comment]
love the test add, thanks for the fix!

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs (hunk 1 above)
Comment (dibarbet) [reply src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs:21]: Add braces seems to use the same implementation as this?  It's a partial
https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Tests/AddBraces/AddBracesTests.cs#L20

Comment (dibarbet) [reply src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs:21]: Tried a few variations of the following test using the new framework, but I don't think its possible.  The new test framework never calls into `IFixAllSpanMappingService`.  Seems like that exists on a higher layer than these tests.  This test always passes even if I remove my fix.

```
[Fact, WorkItem("https://github.com/dotnet/vscode-csharp/issues/6611")]
[Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
public async Task TestFixAll()
{
    var source = """
        using System;
        Console.WriteLine("Hello, World!");

        class [|Program|]
        {
        }
        """;

    var fixedSource = """
        using System;
        Console.WriteLine("Hello, World!");

        internal class Program
        {
        }
        """;

    var test = new VerifyCS.Test
    {
        TestCode = source,
        FixedCode = fixedSource,
        //Options =
        //    {
        //        { CSharpCodeStyleOptions.UnusedValueAssignment, (UnusedValuePreference)option },
        //    },
        LanguageVersion = LanguageVersion.CSharp10,
        // This is not compilable as-is - we have a duplicate Program definition.
        CompilerDiagnostics = CompilerDiagnostics.None,
        BatchFixedCode = fixedSource,
    };

    await test.RunAsync();
}
```

Comment (mavasani) [reply src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs:21]: Ah, my bad. I presume the new testing library doesn’t yet support FixAllInContainingType and FixAllInContainingMember scopes.

**Review by mavasani — APPROVED** [review]
